tests/system/providers/google/bigquery/example_bigquery_dataset.py

```diff
@@ -0,0 +1,94 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for Google BigQuery service testing dataset operations.
+"""
+import os
+from datetime import datetime
+
+from airflow import models
+from airflow.operators.bash import BashOperator
+from airflow.providers.google.cloud.operators.bigquery import (
+    BigQueryCreateEmptyDatasetOperator,
+    BigQueryDeleteDatasetOperator,
+    BigQueryGetDatasetOperator,
+    BigQueryUpdateDatasetOperator,
+)
+from airflow.utils.trigger_rule import TriggerRule
```

Comment:
Is this a required component that authors must remember to include for system tests to work correctly? If so the pre-commit check should verify this is present as well and used below correctly.

Reply:
It's not - we're using trigger rules only if there is teardown-like task. For example if you're creating resource using one task you want to make sure to run task that removes that resource even if some tasks inbetween fails. There may be system tests that don't need it at all.

Reply:
Yeah - see the comments I added above.